# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/FileInfoTImpl.h
# Commit: 7e9d56eb485e
# Full Hash: 7e9d56eb485e06cca2a82fe88e2ae36c2ccd2095
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-04-01 21:26:59
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileManager. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65549
# ==============================================================================

diff -r a753c018a4a8 -r 7e9d56eb485e dom/indexedDB/FileInfoTImpl.h
--- a/dom/indexedDB/FileInfoTImpl.h	Wed Apr 01 09:54:24 2020 +0000
+++ b/dom/indexedDB/FileInfoTImpl.h	Wed Apr 01 09:52:54 2020 +0000
@@ -19,7 +19,7 @@
 template <typename FileManager>
 FileInfoT<FileManager>::FileInfoT(
     const typename FileManager::FileManagerGuard& aGuard,
-    RefPtr<FileManager> aFileManager, const int64_t aFileId,
+    SafeRefPtr<FileManager> aFileManager, const int64_t aFileId,
     const nsrefcnt aInitialDBRefCnt)
     : mFileId(aFileId),
       mDBRefCnt(aInitialDBRefCnt),
@@ -58,8 +58,8 @@
 }
 
 template <typename FileManager>
-FileManager* FileInfoT<FileManager>::Manager() const {
-  return mFileManager;
+FileManager& FileInfoT<FileManager>::Manager() const {
+  return *mFileManager;
 }
 
 template <typename FileManager>
@@ -136,7 +136,7 @@
 
 template <typename FileManager>
 nsCOMPtr<nsIFile> FileInfoT<FileManager>::GetFileForFileInfo() const {
-  const nsCOMPtr<nsIFile> directory = Manager()->GetDirectory();
+  const nsCOMPtr<nsIFile> directory = Manager().GetDirectory();
   if (NS_WARN_IF(!directory)) {
     return nullptr;
   }